app/src/tabs/Statistics/PersonalMessagesChart.tsx

```diff
@@ -0,0 +1,162 @@
+import { styled } from "linaria/react";
+import { lighten } from "polished";
+import React, { useState } from "react";
+import {
+  ResponsiveContainer,
+  PieChart,
+  Pie,
+  Cell,
+  Legend,
+  Sector,
+} from "recharts";
+
+import { User } from "@app/utility/types";
+import { mode, ColorMode } from "@architus/facade/theme/color";
+import { formatNum } from "@architus/lib/utility";
+
+const Styled = {
+  OuterContainer: styled.div`
+    max-width: 100%;
+    height: 100%;
+    display: flex;
+  `,
+  Label: styled.span`
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    word-wrap: break-word;
+    //display: inline-block;
+    max-width: 75%;
+  `,
+  Tooltip: styled.text`
+    font-weight: bold;
+    font-size: 0.75em;
+    ${mode(ColorMode.Dark)} {
+      text-shadow: 0 0 2px rgba(0, 0, 50, 0.5);
+    }
+  `,
+};
+
+export type PersonalMessagesChartProps = {
+  currentUser: User;
+  totalMessages: number;
+  memberCounts: Record<string, number>;
+};
+
+type renderActiveShapeProps = {
+  cx: number;
+  cy: number;
+  innerRadius: number;
+  outerRadius: number;
+  startAngle: number;
+  endAngle: number;
+  fill: string;
+  value: number;
+};
+
+const renderActiveShape = (props: renderActiveShapeProps): JSX.Element => {
```

Comment:
Can you add a doc comment explaining what this component does?

Reply:
I don't know what this component does